### PR TITLE
move run_android_tests.yaml to a reusable workflow (#79182)

### PR DESCRIPTION
**Pull request by jeanschmidt**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #80128

### CODE REVIEW

File: .github/workflows/_run_android_tests.yml [hunk 1]
```diff
@@ -61,10 +45,10 @@ jobs:
           ANDROID_ROOT="/usr/local/lib/android"
           ANDROID_SDK_ROOT="${ANDROID_ROOT}/sdk"
           SDKMANAGER="${ANDROID_SDK_ROOT}/cmdline-tools/latest/bin/sdkmanager"
-          echo "y" | $SDKMANAGER "ndk;21.4.7075529"
+          echo "y" | ${SDKMANAGER} "ndk;21.4.7075529"
```
Comment (kit1980) [inline .github/workflows/_run_android_tests.yml:48]: Why curly braces around single variable here and below?

Comment (jeanschmidt) [reply .github/workflows/_run_android_tests.yml:48]: Somewhat I find it in general more readable, there is some level of discussion on the topic, but in general I see it being in favour of preferring the braces (such as https://bertvv.github.io/cheat-sheets/Bash.html)

File: .github/workflows/trunk.yml [hunk 2]
```diff
@@ -256,3 +256,7 @@ jobs:
     secrets:
       AWS_OSSCI_METRICS_V2_ACCESS_KEY_ID: ${{ secrets.AWS_OSSCI_METRICS_V2_ACCESS_KEY_ID }}
       AWS_OSSCI_METRICS_V2_SECRET_ACCESS_KEY: ${{ secrets.AWS_OSSCI_METRICS_V2_SECRET_ACCESS_KEY }}
+
+  linux-ubuntu-android-tests:
```
Comment (kit1980) [inline .github/workflows/trunk.yml:260]: Let's rename to "android-emulator-build-test". Linux or Ubuntu is not really relevant here.

**Review by kit1980 — APPROVED** [review]

**Comment by kit1980** [comment]
I don't think this was properly merged into master.

**Comment by malfet** [comment]
@jeanschmidt you need to use @pytorchbot to merge PRs (especially ghstack PSs, which target head rather than default branch)

**Comment by jeanschmidt** [comment]
Thanks for the headsup @malfet && @kit1980 =)

**Pull request merged**